lncfg/protocol_legacy_on.go

```diff
@@ -9,15 +9,4 @@ package lncfg
 //
 //nolint:lll
 type LegacyProtocol struct {
-	// CommitmentTweak guards if we should use the old legacy commitment
-	// protocol, or the newer variant that doesn't have a tweak for the
-	// remote party's output in the commitment. If set to true, then we
-	// won't signal StaticRemoteKeyOptional.
-	CommitmentTweak bool `long:"committweak" description:"force node to not advertise the new commitment format"`
```

Comment:
This change also requires us to remove one test iteration in the integration tests that specifically test with `LEGACY` channel types.